server/src/main/java/org/opensearch/telemetry/tracing/WrappedTracer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.opensearch.telemetry.TelemetrySettings;
+import org.opensearch.telemetry.tracing.noop.NoopTracer;
+
+import java.io.IOException;
+
+/**
+ * Wrapper implementation of Tracer. This delegates call to right tracer based on the tracer settings
+ */
+public class WrappedTracer implements Tracer {
```

Comment:
This class should not be public, could you please make it final and package protected?

```
final class WrappedTracer implements Tracer
```

Reply:
Makes sense, addressed in latest commit. Also updated access modifier for other tracing classes.